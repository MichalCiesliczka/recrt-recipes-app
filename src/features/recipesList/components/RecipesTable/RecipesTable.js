// @flow
import React from 'react';

import Table from '@material-ui/core/Table';
import TableHead from '@material-ui/core/TableHead';
import TableBody from '@material-ui/core/TableBody';

import RecipesTableHeader from './elements/RecipesTableHeader';
import RecipesTableRowContainer from './elements/RecipesTableRow';

import type { Recipes } from '../../recipesList.types';
import { availableColumns } from '../../recipesList.constans';

type Props = {
  recipes: Recipes,
};

const RecipesTable = ({ recipes }: Props) => (
  <Table>
    <TableHead>
      <RecipesTableHeader columns={availableColumns} />
    </TableHead>
    <TableBody>
      {recipes
        .map((recipe, index) => (
          <RecipesTableRowContainer
            key={recipe.id}
            recipe={recipe}
            index={index}
            columns={availableColumns}
          />
        ))
      }
    </TableBody>
  </Table>
);

export default RecipesTable;
