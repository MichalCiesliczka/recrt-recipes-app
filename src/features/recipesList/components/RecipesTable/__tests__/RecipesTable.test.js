import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

import RecipesTable from '../RecipesTable';

describe('<RecipesTable />', () => {
  const defaultProps = {
    recipes: [
      {
        id: '1',
        title: 'Test Title',
        description: 'Test Description',
        steps: [
          { name: 'Step 1' },
        ],
        ingredients: [
          { amount: 2, name: 'ingr' },
        ],
      },
    ],
  };

  it('renders without crashing', () => {
    shallow(<RecipesTable {...defaultProps} />);
  });

  it('should render default status', () => {
    const tree = renderer
      .create(
        <Router>
          <RecipesTable {...defaultProps} />
        </Router>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
