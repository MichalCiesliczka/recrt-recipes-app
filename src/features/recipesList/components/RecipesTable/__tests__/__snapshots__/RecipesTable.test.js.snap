// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipesTable /> should render default status 1`] = `
<table
  className="MuiTable-root-1"
>
  <thead
    className="MuiTableHead-root-2"
  >
    <tr
      className="MuiTableRow-root-3 MuiTableRow-head-6"
    >
      <th
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-head-9"
        scope="col"
      >
        No.
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-head-9"
        scope="col"
      >
        Title
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-head-9"
        scope="col"
      >
        Description
      </th>
    </tr>
  </thead>
  <tbody
    className="MuiTableBody-root-16"
  >
    <tr
      className="MuiTableRow-root-3 MuiTableRow-hover-5"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root-8 MuiTableCell-body-10"
        scope={undefined}
      >
        1
      </td>
      <td
        aria-sort={null}
        big={undefined}
        className="MuiTableCell-root-8 MuiTableCell-body-10 Component-column-17 "
        scope={undefined}
      >
        Test Title
      </td>
      <td
        aria-sort={null}
        big="true"
        className="MuiTableCell-root-8 MuiTableCell-body-10 Component-column-17 Component-column--big-18"
        scope={undefined}
      >
        Test Description
      </td>
    </tr>
  </tbody>
</table>
`;
